### PR TITLE
feat!: `acceptInsecureCerts`

### PR DESCRIPTION
**Pull request by sadym-chromium**
BREAKING CHANGE: `BidiServer.createAndStart` signature changed. New optional parameter `options` is added. Breaking change for Puppeteer, while ChromeDriver is not affected, as it uses the Mapper Tab.

When creating Mapper, add possibility to add `acceptInsecureCerts` parameter. Required to provide capability `acceptInsecureCerts`.

Alternative would be handling `session.new` command on the Mapper side, and forward capabilities there. But that would add an extra step to the ChromeDriver - Mapper link.

In order to respect capabilities from both Classic and Bidi, the Mapper launcher has to be refactored, so that it keeps sessions.

### CODE REVIEW

**Review by thiagowfx — APPROVED** [review]

**Comment by thiagowfx** [comment]
LGTM but what is this intended for?

**Comment by sadym-chromium** [comment]
> LGTM but what is this intended for?

Oh, sorry, didn't mean to add you as a reviewer. It's still a draft.

**Comment by sadym-chromium** [comment]
> LGTM but what is this intended for?

extended explanation

File: tests/conftest.py [hunk 1]
```diff
@@ -38,14 +38,26 @@ def local_server(httpserver) -> LocalHttpServer:
...
 
 
+@pytest_asyncio.fixture
+async def websocket(websocket_connection):
+    """Return a websocket with an active BiDi session."""
+    await execute_command(websocket_connection, {
+        "method": "session.new",
+        "params": {}
+    })
+    yield websocket_connection
```
Comment (thiagowfx) [inline tests/conftest.py:58]: return

`yield` is only needed if you have more statements afterwards

File: tests/browsing_context/test_navigate.py [hunk 2]
```diff
@@ -418,3 +419,59 @@ async def test_browsingContext_navigationStarted_browsingContextClosedBeforeNavi
         'id': close_command_id,
         'type': 'success',
     })
+
+
+@pytest.mark.asyncio
+async def test_browsingContext_navigateBadSsl_notNavigated(
+        websocket, context_id):
+    # TODO: make offline.
```
Comment (thiagowfx) [inline tests/browsing_context/test_navigate.py:427]: FYI we already have an off-line fixture

File: tests/conftest.py (hunk 1 above)
Comment (sadym-chromium) [reply tests/conftest.py]: fixed

File: tests/browsing_context/test_navigate.py (hunk 2 above)
Comment (sadym-chromium) [reply tests/browsing_context/test_navigate.py:427]: ack, thanks!

Comment (thiagowfx) [reply tests/browsing_context/test_navigate.py:427]: It's still using the online URL

Comment (thiagowfx) [reply tests/browsing_context/test_navigate.py:427]: bad_ssl_url

**Review by thiagowfx — CHANGES_REQUESTED** [review]
Will re-review in a bit

File: tests/browsing_context/test_navigate.py (hunk 2 above)
Comment (sadym-chromium) [reply tests/browsing_context/test_navigate.py:427]: my bad ~ssl~. Done now

File: tests/conftest.py [hunk 3]
```diff
@@ -38,14 +38,32 @@ def local_server(httpserver) -> LocalHttpServer:
...
     port = os.getenv("PORT", 8080)
     url = f"ws://localhost:{port}"
     async with websockets.connect(url) as connection:
         yield connection
 
 
+@pytest_asyncio.fixture(params=[{"capabilities": {}}])
+async def websocket(request, _websocket_connection):
+    capabilities = request.param['capabilities']
+    """Return a websocket with an active BiDi session."""
```
Comment (thiagowfx) [inline tests/conftest.py:54]: nit: docstring should come first, right after the function definition

File: src/bidiServer/WebSocketServer.ts [hunk 4]
```diff
@@ -30,7 +31,32 @@ const debugInternal = debug('bidi:server:internal');
 const debugSend = debug('bidi:server:SEND ▸');
 const debugRecv = debug('bidi:server:RECV ◂');
 
+type Session = {
+  sessionId: string;
+  // Promised is used to decrease WebSocket handshake latency. If session is set via
```
Comment (thiagowfx) [inline src/bidiServer/WebSocketServer.ts:36]:
```suggestion
  // Promise is used to decrease WebSocket handshake latency. If session is set via
```

File: src/bidiServer/WebSocketServer.ts [hunk 5]
```diff
@@ -126,44 +176,52 @@ export class WebSocketServer {
...
-        }
-      })();
+      }
 
       connection.on('message', async (message) => {
-        // Wait for browser instance to be created.
-        const browserInstance = await browserInstanceDeferred;
-
-        // If |type| is not text, return a error.
+        // If |type| is not text, return error.
```
Comment (thiagowfx) [inline src/bidiServer/WebSocketServer.ts:224]:
```suggestion
        // If type is not text, return error.
```

`||` is spec formatting, not relevant here

File: tests/conftest.py (hunk 3 above)
Comment (sadym-chromium) [reply tests/conftest.py:53]: done

**Review by iKn0wH0w — APPROVED** [review]

**Pull request merged**